.github/workflows/staging-deploy.yml

```diff
@@ -23,10 +28,14 @@ jobs:
         uses: actions/checkout@v3
         with:
           ref: staging
-      - name: remove and replace files 
+      - name: modify files
+      # removes .git folder to avoid conflicts with other repo 
+      # inserts value for key baseurl: "/website-staging" 
+      # appends the repository key and value to the end of the _config file
         run: |
           rm -rf .git
-          mv _config_staging.yml _config.yml 
+          sed 's/\(^baseurl:\).*/\1 "\/website-staging" /' _config.yml 
```

Comment:
I would have written this `sed 's/^baseurl: ""/baseurl: "\/website-staging" /' _config.yml`

I thought it odd that the regex was capturing, but this is not a strength for me.

Reply:
capturing in general? or something specific?

Reply:
I should have said "capturing a string literal" without any regex flags in it. (Technically you're not doing that, but the ^ at the start of the capture is close-- it doesn't prevent doing it without capture.)

Reply:
ah ok - the capture `(^baseurl:\)` looks for the first baseurl at the beginning of a line, then `.*` adds the new string after it. `/1` represents the captured string `baseurl:`. I did it this way to "future proof" as much as possible. If the `_config` file was ever updated, like a value was given to `baseurl: "/something-new"` the staging deploy would not break. I would just override what ever value is set there.